src/app/src/main/java/com/example/bookmark/EditProfileActivity.java

```diff
@@ -1,17 +1,174 @@
+            if (hasFocus) {
+                emailAddressLayout.setError(null);
+            } else {
+                if (emailAddressEditText.getText().toString().equals("")) {
+                    validateEditTextEmpty(emailAddressEditText, emailAddressLayout);
+                } else {
+                    checkIfEditTextValidEmail(emailAddressEditText, emailAddressLayout);
+                }
+            }
+        });
+
+        phoneNumberEditText.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) {
+                phoneNumberLayout.setError(null);
+            } else {
+                if (phoneNumberEditText.getText().toString().equals("")) {
+                    validateEditTextEmpty(phoneNumberEditText, phoneNumberLayout);
+                } else {
+                    checkIfEditTextValidPhoneNumber(phoneNumberEditText, phoneNumberLayout);
+                }
+            }
+        });
+
+        firstNameEditText.setOnFocusChangeListener(new EmptyTextFocusListener(firstNameEditText, firstNameLayout));
+        lastNameEditText.setOnFocusChangeListener(new EmptyTextFocusListener(lastNameEditText, lastNameLayout));
+
+        phoneNumberEditText.addTextChangedListener(new PhoneNumberFormattingTextWatcher());
+        phoneNumberEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                checkIfEditTextValidPhoneNumber(phoneNumberEditText, phoneNumberLayout);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+
+        emailAddressEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                checkIfEditTextValidEmail(emailAddressEditText, emailAddressLayout);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+
+        // pre populate fields with user current user info
+        SharedPreferences sharedPref = this.getSharedPreferences("LOGGED_IN_USER", Context.MODE_PRIVATE);
+        loggedUsername = sharedPref.getString("USER_NAME", "ERROR_NO_USER");
+        if (loggedUsername.equals("ERROR_NO_USER")) {
+            DialogUtil.showErrorDialog(this, new Exception(loggedUsername));
+        } else {
+            prepopulateTextFields(loggedUsername);
+        }
+    }
+
+    private void prepopulateTextFields(String username) {
+        FirebaseProvider.getInstance().retrieveUserByUsername(username, user -> {
```

Comment:
This is no longer how to access the FirebaseProvider. After https://github.com/CMPUT301F20T01/Bookmark/pull/77 is merged we will use `StorageService storageService = StorageServiceLocator.getInstance().getStorageService();` to abstract this out and enabled intent testing.